### PR TITLE
Add labeledUrl user option type

### PR DESCRIPTION
**Pull request by joshuaruesweg**

### CODE REVIEW

File: wcfsetup/install/lang/de.xml [hunk 1]
```diff
@@ -3084,6 +3084,9 @@ Wenn {if LANGUAGE_USE_INFORMAL_VARIANT}du{else}Sie{/if} unter „Konfiguration 
 		<item name="wcf.acp.user.option.editable.2"><![CDATA[Administrator]]></item>
 		<item name="wcf.acp.user.option.editable.3"><![CDATA[Eigentümer und Administrator]]></item>
 		<item name="wcf.acp.user.option.editable.6"><![CDATA[Eigentümer bei Erstanmeldung und Administrator]]></item>
+		<item name="wcf.acp.user.option.labeledUrl"><![CDATA[URL]]></item>
+		<item name="wcf.acp.user.option.labeledUrl.description"><![CDATA[Wenn der Feldtyp „labeledUrl“ ist, muss hier die URL hinterlegt werden, auf die verlinkt werden soll. {if LANGUAGE_USE_INFORMAL_VARIANT}Nutze{else}Nutzen Sie{/if} <span class="inlineCode">%s</span> als Platzhalter für die Eingabe des Benutzers.]]></item>
```
Comment (joshuaruesweg) [inline wcfsetup/install/lang/de.xml:3088]: Any idea for a better description?

File: wcfsetup/install/files/lib/acp/form/UserOptionAddForm.class.php [hunk 2]
```diff
@@ -235,6 +243,10 @@ protected function setDefaultOutputClass() {
 			if ($this->optionType == 'URL') {
 				$this->outputClass = 'wcf\system\option\user\URLUserOptionOutput';
 			}
+			
+			if ($this->optionType == 'labeledUrl') {
+				$this->outputClass = 'wcf\system\option\user\LabeledUrlUserOptionOutput';
```
Comment (TimWolla) [inline wcfsetup/install/files/lib/acp/form/UserOptionAddForm.class.php:248]:
```suggestion
				$this->outputClass = LabeledUrlUserOptionOutput::class;
```
?

File: wcfsetup/install/files/lib/system/option/user/LabeledUrlUserOptionOutput.class.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	 */
+	public function getOutput(User $user, UserOption $option, $value) {
+		return StringUtil::getAnchorTag(self::getURL($option, $value), $value, true, true);
+	}
+	
+	/**
+	 * Formats the URL.
+	 */
+	private static function getURL(UserOption $option, string $value): string {
+		return sprintf($option->labeledUrl, urlencode($value));
```
Comment (TimWolla) [inline wcfsetup/install/files/lib/system/option/user/LabeledUrlUserOptionOutput.class.php:28]:
```suggestion
		return sprintf($option->labeledUrl, rawurlencode($value));
```

File: wcfsetup/install/lang/de.xml [hunk 4]
```diff
@@ -3084,6 +3084,9 @@ Wenn {if LANGUAGE_USE_INFORMAL_VARIANT}du{else}Sie{/if} unter „Konfiguration 
...
+		<item name="wcf.acp.user.option.labeledUrl.error.invalid"><![CDATA[Die URL enthält nicht den Platzhalter <span class="inlineCode">%s</span>.]]></item>
```
Comment (TimWolla) [inline wcfsetup/install/lang/de.xml:3089]:
```suggestion
		<item name="wcf.acp.user.option.labeledUrl.error.invalid"><![CDATA[Die URL enthält den Platzhalter <span class="inlineCode">%s</span> nicht.]]></item>
```

**Review by TimWolla — CHANGES_REQUESTED** [review]

File: wcfsetup/install/lang/de.xml (hunk 4 above)
Comment (joshuaruesweg) [reply wcfsetup/install/lang/de.xml:3089]: The suffixed "nicht" sounds very strange to me. I would therefore not implement your proposal.

Comment (Gravatronics) [reply wcfsetup/install/lang/de.xml:3089]: `Den Platzhalter <span class="inlineCode">%s</span> fehlt in der URL.`?

File: wcfsetup/install/files/acp/templates/userOptionAdd.tpl [hunk 5]
```diff
@@ -139,6 +139,23 @@
 				</dd>
 			</dl>
 			
+			<dl{if $errorField == 'labeledUrl'} class="formError"{/if}>
+				<dt><label for="selectOptions">{lang}wcf.acp.user.option.labeledUrl{/lang}</label></dt>
```
Comment (TimWolla) [inline wcfsetup/install/files/acp/templates/userOptionAdd.tpl:143]:
```suggestion
				<dt><label for="labeledUrl">{lang}wcf.acp.user.option.labeledUrl{/lang}</label></dt>
```

**Review by TimWolla — APPROVED** [review]
Other than that it LGTM now.

**Pull request merged**